### ISSUE TITLE
BUG: Files in SuperLU under LGPL license

### ISSUE THREAD

**Issue by hawkinsp** [scipy/scipy#21234]
### Describe your issue.

At least historically, scipy didn't allow LGPL code (https://scipy.github.io/old-wiki/pages/License_Compatibility.html), although the current documentation is a bit less clear on that point (https://docs.scipy.org/doc/scipy/dev/core-dev/index.html#licensing)

A [commit which updated SuperLU](https://github.com/scipy/scipy/commit/f4bbb25d2931a91dbad4e6528f18648d02b5d697#diff-322b45f4593647eb176df1d6e023165592a1b9db917e6dacddd6f750b38fb86f) updated a file which now appears to be under an LGPL license (scipy/sparse/linalg/_dsolve/SuperLU/SRC/colamd.c). The top of the file asserts the file has a BSD license, but if you look lower it seems an LGPL license applies to this code.

This makes scipy code a bit harder to work with from a license perspective for end users. Was this intentional and could it be fixed?

If nothing else, I suspect the LICENSE file in the distributed SciPy wheels should mention the LGPL license on the SuperLU code, which it currently does not.

@rgommers 

### Reproducing Code Example

```python
n/a
```

### Error message

```shell
n/a
```

### SciPy/NumPy/Python version and system information

```shell
1.12+
```

**Comment by lucascolley** [comment]
cc @xiaoyeli

**Comment by ilayn** [comment]
We have contacted the author previously see #15936 and also a few other places and if I remember correctly had explicit permission on this issue. I don't think it is a problem but would be nice to make it obvious.

**Comment by rgommers** [comment]
Yes, I believe we previously had permission, _however_ this is still an issue for several reasons:

1. For SciPy itself, having permission for a previous version of the COLAMD code doesn't automatically mean that we can incorporate an updated version without a new permission.
2. For SuperLU, there's a problem in that file with conflicting license info in the same file - the LGPL mention should be removed
3. It'll be a practical problem for packagers (and us) because Linux distros and other redistributors have license checkers that are going to trigger on the in-tree mention of LGPL, so it causes unnecessary work for packagers to have that LGPL mention there.

Note that `colamd.c|h` are explicitly BSD-licensed: https://github.com/DrTimothyAldenDavis/SuiteSparse/blob/26ababc7f3af725c5fb9168a1b94850eab74b666/COLAMD/README.txt

And have SPDX license expressions in every file now:
https://github.com/DrTimothyAldenDavis/SuiteSparse/blob/26ababc7f3af725c5fb9168a1b94850eab74b666/COLAMD/Source/colamd.c#L7

So there should not be a licensing problem here, just an "erroneous mention of LGPL" problem that has to be fixed. I'll open an issue upstream right now.

**Comment by xiaoyeli** [comment]
resolved.

**Comment by rgommers** [comment]
Thank you @hawkinsp and @xiaoyeli!

We should have a separate PR with just the relevant diff from the fix in SuperLU, so we can backport it easily to 1.14.x. @dschmitz89 do you want to do it, or should I?

**Comment by dschmitz89** [comment]
> Thank you @hawkinsp and @xiaoyeli!
> 
> We should have a separate PR with just the relevant diff from the fix in SuperLU, so we can backport it easily to 1.14.x. @dschmitz89 do you want to do it, or should I?

Hey guys, I want to look into it but if 1.14.1 is coming soon, I cannot promise that I will make it in time.

**Comment by rgommers** [comment]
Thanks @dschmitz89. I think it'd be nice to resolve asap, so I had a look and opened gh-21274 for it just now.
